src/impl_ops.rs

```diff
@@ -218,6 +301,23 @@ impl<'a, S, D> $trt<&'a ArrayBase<S, D>> for $scalar
         })
     }
 }
+
+/// Perform elementwise
+/// between the scalar `self` and array `rhs`,
+/// and return the result as a new `Array`.
```

Comment:
Not important, but just something I saw. Allegedly this doc is not visible in rustdoc, but that comment may be outdated now(!). then we want that `#[doc=$doc]` back to make a complete sentence.

Reply:
I would especially discourage these impl blocks. We have to spend a lot of impl blocks on the arithmetic ops with scalars on the *left hand side*. Part of me want to just remove everything with left hand side scalars, because it can never(?) be properly general because of how we need to express it as an `impl .. for $scalar` (for a specific single type).

Reply:
I've removed this impl block in the most recent commit because it's not necessary.

> Part of me want to just remove everything with left hand side scalars, because it can never(?) be properly general because of how we need to express it as an `impl .. for $scalar` (for a specific single type).

I understand the objection, and AFAIK it will never be possible to write these impls for generic scalars even after [re_rebalance_coherence](https://doc.rust-lang.org/unstable-book/language-features/re-rebalance-coherence.html). However, IMO supporting left-hand-side scalars is important, even if it's just `f32`/`f64`. I use these impls all the time in my code. They are especially important for non-commutative ops, where it's not straightforward to just move the scalar to the other side.